### ISSUE TITLE
[ENHANCEMENT]

### ISSUE THREAD

**Issue by abinav369** [ibus/ibus-m17n#91]
**Is your feature request related to a problem? Please describe.**
The TamilNet'99 nor Tamil inscript keyboard isn't what we use in our state. Here the standard is tamil99 keyboard layout which is missing in this engine.

**Describe the solution you'd like**
Add Thamizh99 keyboard layout to the engine

**Describe alternatives you've considered**
Keyman has this layout, which can also be added to Ibus. It works fine for Debian based distros(I have checked it personally) but it is not the case with arch. It works, but only two letters at the max can be typed in Tamil then it switches to English even though the layout is Thamizh99

**Additional context**
![image](https://github.com/user-attachments/assets/1854e4a1-c0f2-42ed-a6a3-8d36b7a3e2bc)
![image](https://github.com/user-attachments/assets/89fb2a53-abb1-4a77-aab6-dfbf760b5427)

**Comment by mike-fabian** [comment]
You are mentioning a TamilNet'99 layout and I am not sure what you mean by that. Maybe you mean the layout from xkeyboard-config?

The file `/usr/share/X11/xkb/symbols/in` which is part of xkeyboard-config:

```
$ rpm -qf  /usr/share/X11/xkb/symbols/in
xkeyboard-config-2.42-2.fc41.noarch
```

contains a `tamilnet` layout:

```
$ grep -B1 -A 10 'xkb_symbols "tamilnet"'  /usr/share/X11/xkb/symbols/in
partial alphanumeric_keys
xkb_symbols "tamilnet" {

    // Description: A keymap based on the TamilNet'99 typewriter keyboard
    // Encoding: Unicode (http://www.unicode.org)
    // Author: Thuraiappah Vaseeharan <vasee@ieee.org>
    // Modifed by: Malathi S <malathiramya@gmail.com>
    // Secondary contact: Sri Ramadoss M <amachu@au-kbc.org>
    // Date  : Fri Sep 4 11:32:00 CST 2009

    name[Group1]= "Tamil (TamilNet '99)";
```

This has nothing to do with ibus-m17n though, this is just a keyboard layout for Xorg and Wayland.


ibus-m17n (and ibus-typing-booster) use the input methods from m17n-db. I.e. they can use this one for example:

https://git.savannah.nongnu.org/cgit/m17n/m17n-db.git/tree/MIM/ta-tamil99.mim

So if you use ibus-m17n or ibus-typing-booster, you can use this `ta-tamil99.mim` from m17n-db.

I don’t know how that compares to the Tamil99 layout in Keyman. Maybe it is similar?

Can you try it please and tell me whether `ta-tamil99.mim` does what you want?

**Comment by abinav369** [comment]
Since I am a noob, I might not have asked the right question under the right repository. Sorry for that. But as you suggested, If I install `m17n-db`, my issue will be solved as `ta-tamil99.mim` input method is what I was looking for.

Now I face an issue while installing `m17n-db`.
I am not sure if I can continue in this repo. Anyways,

```
$ ./configure
configure: error: cannot find required auxiliary files: ltmain.sh
```

**Comment by mike-fabian** [comment]
@abinav369 Doesn't Arch have packages for m17n-db and m17n-lib? To use the `ta-tamil99.mim` input method, you need to install m17n-lib, m17n-db, ibus, and ibus-m17n. I think there are packages for all of this for arch: 

https://archlinux.org/packages/extra/x86_64/m17n-lib/
https://archlinux.org/packages/extra/any/m17n-db/
https://aur.archlinux.org/packages/ibus-m17n
https://archlinux.org/packages/extra/x86_64/ibus/

I don't know much about Arch Linux and how package management works there.

Of course you can also install from source if you want to. But as you describe your self as a noob (which probably means you don't have very much experience), you might encounter some difficulties. If you want, I am willing to help you though if you really want to do that (But using packages is probably easier!).

If you start with the release tarball, available here:

http://download.savannah.nongnu.org/releases/m17n/m17n-db-1.8.9.tar.gz

then `ltmain.sh` should be there as it is part of the release tarball:

```
$ tar tvf m17n-db-1.8.9.tar.gz | grep ltmain
-rw-r--r-- 10030/10030  332808 2024-11-14 13:16 m17n-db-1.8.9/ltmain.sh
```

So you should be able to run `./configure` just fine. 

If you start with a `git clone`, (see: https://savannah.nongnu.org/git/?group=m17n how to get a git clone), then there is no ltmain.sh yet and there is no configure yet. You need to run `./bootstrap.sh` which will create configure and ltmain.sh and other  necessary files. But `./bootstrap.sh` requires several other tools like autoreconf, automake, and others. If you have little experience, you might find that difficult. Starting with release tarball is easier as more needed generated files are already included and you need fewer tools for building and installing.

Please ask more if you have problems, I'll try to help you!

**Comment by abinav369** [comment]
I was so dumb not to check the official package manager. Yup `m17n-db` package can be installed using `pacman`.
Finally, I have installed the input method which I wanted. But the thing is, `m17n-db` package is not required. `tamil99` input method was available in `ibus-m17n` package itself which I did install from aur.

In conclusion, installing `ibus` with `ibus-m17n` package does the job.

மிக்க நன்றி (Thank you so much in Thamizh (pronounced as Thamil) ).
